src/librustc_lint/unused.rs

```diff
@@ -65,9 +65,8 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnusedResults {
             ty::Adt(def, _) => {
                 if def.variants.is_empty() {
                     return;
-                } else {
-                    check_must_use(cx, def.did, s.span, "")
                 }
+                check_must_use(cx, def.did, s.span, "")
```

Comment:
This still seems to be unrelated to this change. (It's a good change, but not related to this commit.)

Reply:
OK, I backed this out.

Reply:
Thanks! (Please do submit this cleanup separately.)